src/Microsoft.Azure.WebJobs.Host/Hosting/WebJobsExtensionBuilderExtensions.cs

```diff
@@ -72,6 +76,11 @@ public static class WebJobsExtensionBuilderExtensions
             builder.Services.AddSingleton<IConfigureOptions<TOptions>>(p =>
                 new WebJobsExtensionOptionsConfiguration<TOptions>(p.GetService<IConfiguration>(), builder.ExtensionInfo.ConfigurationSectionName, configure));
 
+            if (typeof(IOptionsFormatter).IsAssignableFrom(typeof(TOptions)))
+            {
+                builder.Services.AddSingleton<IExtensionOptionsInfo>(p => new ExtensionOptionsInfo(builder.ExtensionInfo, p.GetService<IOptions<TOptions>>().Value as IOptionsFormatter));
```

Comment:
It's possible ConfigureOptions to be called multiple times for the same options type, with different configure actions. In that case, above we correctly register multiple IConfigureOptions instances so at build time they'll be applied in order.

For our new IExtensionOptionsInfo instances however, they won't vary based on the action. Can you test what happens here - perhaps update your unit test so that for one of your test extensions, you call ConfigureOptions directly multiple times, with different actions that set options properties. I'm assuming we'll end up with multiple duplicate IExtensionOptionsInfos registered which likely doesn't cause any problems, however it is strange. I wonder if there's a way we can avoid that.

Reply:
This will be equivalent of registering enumerables. You'll have a collection of them that you can get by taking a dependency on `IEnumerable<IExtensionOptionsInfo>`

Reply:
Right - in both cases, you'll end up with a collection of separate instances, each of which is effectively the same. Likely doesn't happen that often, and not a huge concern since consumer can just pick one, I just didn't know if there was a simple pattern we could employ here to only register ONE for a given TOptions/builder pair.

Reply:
Update as we talked.